### PR TITLE
[presstv] Added extractor for PressTV.

### LINKED ISSUE #7060: report: Unsupported URL (PressTV)
Thank you for this great tool. Works like a charm for me.

I would like to urge you to support this important URL.

Here ist my complete log:

[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['--verbose', 'http://www.presstv.ir/Video/2015/10/04/431915/Max-Igan-Press-TV-Face-to-Face']
[debug] Encodings: locale UTF-8, fs UTF-8, out None, pref UTF-8
[debug] youtube-dl version 2014.08.05
[debug] Python version 2.7.9 - Linux-3.16.0-4-amd64-x86_64-with-debian-8.2
[debug] Proxy map: {}
[generic] Max-Igan-Press-TV-Face-to-Face: Requesting header
WARNING: Falling back on generic information extractor.
[generic] Max-Igan-Press-TV-Face-to-Face: Downloading webpage
[generic] Max-Igan-Press-TV-Face-to-Face: Extracting information
ERROR: Unsupported URL: http://www.presstv.ir/Video/2015/10/04/431915/Max-Igan-Press-TV-Face-to-Face; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/youtube_dl/extractor/generic.py", line 457, in _real_extract
    doc = parse_xml(webpage)
  File "/usr/lib/python2.7/dist-packages/youtube_dl/utils.py", line 1417, in parse_xml
    return xml.etree.ElementTree.XML(s.encode('utf-8'), **kwargs)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1300, in XML
    parser.feed(text)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1642, in feed
    self._raiseerror(v)
  File "/usr/lib/python2.7/xml/etree/ElementTree.py", line 1506, in _raiseerror
    raise err
ParseError: mismatched tag: line 24, column 2
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/youtube_dl/YoutubeDL.py", line 516, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 170, in extract
    return self._real_extract(url)
  File "/usr/lib/python2.7/dist-packages/youtube_dl/extractor/generic.py", line 752, in _real_extract
    raise ExtractorError('Unsupported URL: %s' % url)
ExtractorError: Unsupported URL: http://www.presstv.ir/Video/2015/10/04/431915/Max-Igan-Press-TV-Face-to-Face; please report this issue on https://yt-dl.org/bug . Be sure to call youtube-dl with the --verbose flag and include its complete output. Make sure you are using the latest version; type  youtube-dl -U  to update.

### PR DESCRIPTION
**Pull request by Phaeilo**
Fixes #7060

### CODE REVIEW

**Comment by yan12125** [comment]
Since #8497 landed, please move changes in ```youtube_dl/extractors/__init__.py``` to ```youtube_dl/extractors/extractors.py```. Check step 5 of the new [developer instructions](https://github.com/rg3/youtube-dl/blob/master/README.md#adding-support-for-a-new-site) for more information.

File: youtube_dl/extractor/presstv.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        'md5': '5d7e3195a447cb13e9267e931d8dd5a5',
+        'info_dict': {
+            'id': '459911',
+            'ext': 'mp4',
+            'title': 'Organic mattresses used to clean waste water',
+            'upload_date': '20160409',
+            'thumbnail': 'http://media.presstv.com/photo/20160409/41719129-76fa-4372-a09d-bf348278eb5d.jpg',
+            'description': ('A trial program at an Australian sewerage treatment facility hopes to change '
+                            'the way waste water is treated by using plant mattresses to reduce chemical '
+                            'and electricity use.')
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: Usually we use ```md5:(md5 of the long text)``` for such cases.

File: youtube_dl/extractor/presstv.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            }, {
+                'url': base_url + video_url.replace(".mp4", "_low400.mp4"),
+                'format': '360p mp4',
+                'format_id': '360p'
+            }, {
+                'url': base_url + video_url.replace(".mp4", "_low200.mp4"),
+                'format': '180p mp4',
+                'format_id': '180p'
+            }
+        ]
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: Use a for loop to eliminate duplicated codes.

File: youtube_dl/extractor/presstv.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        video_url = self._html_search_regex(r'<input type="hidden" id="inpPlayback" value="([^"]+)" />', webpage,
+                                            'Video URL')
+
+        # build list of available formats
+        # specified in http://www.presstv.ir/Scripts/playback.js
+        base_url = 'http://192.99.219.222:82/presstv'
+        formats = [
+            {
+                'url': base_url + video_url,
+                'format': '1080p mp4',
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: This field is not necessary - it does not provide more information than what ```format_id``` and ```ext```. Also, height values should be placed in the field ```height```.

File: youtube_dl/extractor/presstv.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            }
+        ]
+        formats.reverse()
+
+        # extract video metadata
+        title = self._html_search_meta('title', webpage, 'Title', True)
+        title = title.partition('-')[2].strip()
+
+        thumbnail = self._html_search_meta('og:image', webpage, 'Thumbnail', True)
+        description = self._html_search_meta('og:description', webpage, 'Description', True)
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: Use ```self._og_search_*``` functions here.

File: youtube_dl/extractor/presstv.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+        year = str_to_int(self._search_regex(PressTVIE._VALID_URL, url, 'Upload year', group='y'))
+        month = str_to_int(self._search_regex(PressTVIE._VALID_URL, url, 'Upload month', group='m'))
+        day = str_to_int(self._search_regex(PressTVIE._VALID_URL, url, 'Upload day', group='d'))
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: In this case use snippets like: (no testsed)
```
mobj = re.match(self._VALID_URL, url)
video_id = mobj.group('id')
year = mobj.group('y')
month = mobj.group('m')
day = mobj.group('d')
```

File: youtube_dl/extractor/presstv.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    _TEST = {
+        'url': 'http://www.presstv.ir/Detail/2016/04/09/459911/Australian-sewerage-treatment-facility-/',
+        'md5': '5d7e3195a447cb13e9267e931d8dd5a5',
+        'info_dict': {
+            'id': '459911',
+            'ext': 'mp4',
+            'title': 'Organic mattresses used to clean waste water',
+            'upload_date': '20160409',
+            'thumbnail': 'http://media.presstv.com/photo/20160409/41719129-76fa-4372-a09d-bf348278eb5d.jpg',
```
Comment (yan12125) [inline youtube_dl/extractor/presstv.py]: Usually this field does use full URLs. Instead ```'re:^https?://.*\.jpg$'``` as described in https://github.com/rg3/youtube-dl#adding-support-for-a-new-site.

**Comment by yan12125** [comment]
Thanks merged. Would you like to be listed in [AUTHORS](https://github.com/rg3/youtube-dl/blob/master/AUTHORS)? If so under what name?

**Comment by Phaeilo** [comment]
Feel free to list me as Philip Huppert.

**Pull request merged**